MapboxCoreNavigation/RouteAlert.swift

```diff
@@ -0,0 +1,88 @@
+import Foundation
+import CoreLocation
+import MapboxNavigationNative
+import MapboxDirections
+
+/**
+ `RouteAlert` encapsulates information about various incoming events. Common attributes like location, distance to the event, length and other is provided for each POI, while specific meta data is supplied via `alert` property.
+ */
+public struct RouteAlert {
+    public enum Alert {
+        case incident(Incident)
+        case tunnel(Tunnel)
+        case borderCrossing(BorderCrossing)
+        case tollCollection(TollCollection)
+        case serviceArea(RestStop)
+        case restrictedArea
+    }
```

Comment:
This enumeration and its cases should be documented, since the public `alert` property uses them. It can be just a perfunctory documentation comment, such as “Incident with the given incident details” for `incident`.

Reply:
Doc comments added